### ISSUE TITLE
With the HTTPie integration, is there a way to send "ext"?

### ISSUE THREAD

**Issue by jbisotti** [mozilla-services/requests-hawk#19]
I've been trying to find a good way to test our APIs, which use Hawk, and I came across this HTTPie integration. Our APIs rely on some extra data ("ext") in the Hawk header. Is it possible to provide that using this?

Thank you.

**Comment by Natim** [comment]
I don't think so but it should be possible to add an environment variable to support it.

**Comment by jtmaclachlan** [comment]
There is a pull request open to allow for an `ext` parameter to be passed in the Hawk header, just needs to be reviewed and merged.

**Comment by jhwarner** [comment]
The `ext` change has been merged, but I don't see a way to use that from within HTTPie. Is there a way?

**Comment by Natim** [comment]
Hello @jhwarner,

As of today, the Httpie integration is [really lightweight](https://github.com/mozilla-services/requests-hawk/blob/master/requests_hawk/__init__.py#L146)

Do you have an API in mind you would like to use in that case? Would a env variable works for you?

You can register your own plugin like so too:

```python
    import os
    from httpie.plugins import AuthPlugin

    class HawkPlugin(AuthPlugin):

        name = 'Hawk Auth'
        auth_type = 'hawk'
        description = ''

        def get_auth(self, username, password):
            ext = os.getenv("HAWK_EXT_PAYLOAD")
            if password == '':
                return HawkAuth(hawk_session=username, ext=ext)
            return HawkAuth(id=username, key=password, ext=ext)

```
